AngularApp/src/app/shared/providers/resource.service.provider.ts

```diff
@@ -1,19 +1,30 @@
 import { StartupService } from "../services/startup.service";
-import { ResourceType } from "../models/resources";
+import { ArmResource } from "../models/resources";
 import { SiteService } from "../services/site.service";
-import { DiagnosticApiService } from "../services/diagnostic-api.service";
-import { ActivatedRoute } from "@angular/router/src/router_state";
-import { startTimeRange } from "@angular/core/src/profile/wtf_impl";
 import { AseService } from "../services/ase.service";
+import { ResourceService } from "../services/resource.service";
+import { ObserverService } from "../services/observer.service";
 
-export let ResourceServiceFactory = (startupService: StartupService, siteService: SiteService, aseService: AseService) => {
-    let type: ResourceType = startupService.getResourceType();
+export let ResourceServiceFactory = (startupService: StartupService, observerService: ObserverService) => {
+    let resource: ArmResource = startupService.getResourceInfo();
+    let type = `${resource.provider}/${resource.resourceTypeName}`
+
+    let service: ResourceService;
 
     switch (type) {
-        case ResourceType.AppServiceEnvironment:
-            return aseService;
-        case ResourceType.Site:
+        case 'Microsoft.Web/hostingEnvironments':
+            service = new AseService(observerService);
+            service.initialize(resource);
```

Comment:
can the initialize(resource) be moved inside the service constructor

Reply:
That would be ideal, it is tricky because anything that is injected into a service needs to be an injection service, There is a way we could do it I think, I may update it.

Reply:
I tried to do it that way and got errors

Reply:
I am going to try to fix this before committing